PowerShellTools.HostService/ServiceManagement/Debugging/PowerShellDebuggingService.cs

```diff
@@ -160,26 +188,29 @@ public void AttachToRunspace(uint pid)
             // Enter into to-attach process which will swap out the current runspace
             using (_currentPowerShell = PowerShell.Create())
             {
+                _attachRequestEvent.Reset();
+                DebugScenario preScenario = GetDebugScenario();
                 _currentPowerShell.Runspace = _runspace;
                 _currentPowerShell.AddCommand("Enter-PSHostProcess").AddParameter("Id", pid.ToString());
                 _currentPowerShell.Invoke();
 
-                // wait for invoke to finish swapping the runspaces
-                _attachRequestEvent.WaitOne(5000);
-
-                // make sure that the semaphore didn't just time out
-                if (GetDebugScenario() != DebugScenario.LocalAttach)
+                // wait for invoke to finish swapping the runspaces if you are attaching to a local process
+                if (preScenario != DebugScenario.RemoteSession)
                 {
-                    MessageBox.Show(Resources.ProcessAttachFailErrorBody, Resources.ProcessAttachFailErrorTitle, MessageBoxButton.OK, MessageBoxImage.Warning);
-                    ServiceCommon.Log("Failed to attach to running process.");
-                    return;
+                    _attachRequestEvent.WaitOne(5000);

```

Comment:
Since we start to use this arbitrary 5000 multiple places, let's put it in a const